packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -1026,8 +1000,11 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         parentHandle: string | undefined,
```

Comment:
I think I'm following most of the logic, but it's easy to get lost. Could you please write a small documentation in the header of this function on all the things it does, why and how

Reply:
That should also help to realize what tests we need to write - ideally we can write UTs targeting behaviors of this function (maybe need to refactor code to make it available without bigger class it's hosted now, i.e. move into smaller class maybe that deals with all the caching things?)

Reply:
yes, I will add that.

Reply:
Made a separate class for this.

Reply:
Added documentation.